crates/pallets/starknet/src/lib.rs

```diff
@@ -995,5 +1000,147 @@ pub mod pallet {
+                    TransactionValidityError::Unknown(Custom(1_u8))
+                })?, // TODO: implement account id to match starknet and use who
+                call_type: blockifier::execution::entry_point::CallType::Call,
+            };
+            let mut execution_context = ExecutionContext::default(); // TODO: check if it needs a real value.
+            let account_ctx = AccountTransactionContext::default(); // TODO: check if it needs a real value.
+            let block_ctx = BlockContext {
+                chain_id: ChainId("SN_GOERLI".to_string()), // TODO: Make it configurable ?
+                block_number: BlockNumber(block.header().block_number.as_u64()),
+                block_timestamp: BlockTimestamp(block.header().block_timestamp),
+                sequencer_address: ContractAddress::try_from(
+                    StarkFelt::new(block.header().sequencer_address).map_err(|_| {
+                        log!(error, "Couldn't convert sequencer address to StarkFelt");
+                        TransactionValidityError::Unknown(Custom(0_u8))
+                    })?,
+                )
+                .map_err(|_| {
+                    log!(error, "Couldn't convert StarkFelt to ContractAddress");
+                    TransactionValidityError::Unknown(Custom(1_u8))
+                })?,
+                cairo_resource_fee_weights: HashMap::default(), // TODO: Use real weights
+                fee_token_address,
+                invoke_tx_max_n_steps: 1000000, // TODO: Make it configurable
+                validate_max_n_steps: 1000000,  // TODO: Make it configurable
+                gas_price: 0,                   // TODO: Use block gas price
+            };
+            match fee_transfer_call.execute(
+                state,
+                &mut ExecutionResources::default(),
+                &mut execution_context,
+                &block_ctx,
+                &account_ctx,
+            ) {
+                Ok(v) => {
+                    log!(info, "Fees executed successfully: {:?}", v); // TODO: remove this log
+                }
+                Err(e) => {
+                    log!(error, "Fees execution failed: {:?}", e);
+                    return Err(TransactionValidityError::Invalid(Payment));
+                }
+            }
+            Pallet::<T>::apply_state_diffs(state).map_err(|_| {
+                log!(error, "Couldn't apply the state diffs");
+                TransactionValidityError::Unknown(Custom(3_u8))
+            })?;
+            Ok(())
+        }
+    }
+    pub struct StarknetFee;
+    impl<T: Config> OnChargeTransaction<T> for StarknetFee {
+        /// The underlying integer type in which fees are calculated.
+        type Balance = u128;
+
+        type LiquidityInfo = U256;
+
+        /// Before the transaction is executed the payment of the transaction fees
+        /// need to be secured.
+        ///
+        /// Note: The `fee` already includes the `tip`.
```

Comment:
add rust doc for Arguments and Returns

Reply:
Commenting the inside of a function yes but describing the parameters doesn’t imo as it’s a trait so all implementations will receive the same params. Anyway here the trait is not commented so I’ll do it but in general I don’t think we should

Reply:
I strongly disagree. Each implementation might have important information that needs to be present in the rust doc.

Reply:
And i also strongly disagree you need to send the same error type so it's just a detail of implementation. I strongly believe that the rust doc should be on the trait (even rust analyzer checks for the doc on the trait) and the implementation details of the implementation should be inside the function